Fix 84 - Apply Cookiecutter style to Circup Repo

askpatrickw opened the pull request:
Fixes #84
This adds pre-commit and removed redundant commands from Makefile
Once pre-commit was in place, I fixed or ignored all errors, added licenses, etc..

kattni reviewed (commented):
Looks good to me, but I'm unfamiliar with the circup situation, so I'll leave this for a least a bit for others to review!

jepler commented on make/make.py:
```diff
@@ -256,14 +229,16 @@ def help():
     try:
         function = _exported[command]
     except KeyError:
-        raise RuntimeError("No such command: %s" % command)
+        raise RuntimeError("No such command: %s" % command) from None
```
should probably be `except KeyError as e: ... raise ... from e`

askpatrickw replied:
Yeah, I went cheap on those errors because I wasn't sure if any of that was going to survive when made this whole repo look more like a library repo. I'll re-work that when I do the github action work.

jepler reviewed (changes requested):
Besides my requested change, I don't see a change to .github/actions to run pre-commit during the CI process (& remove them from the usual build.yml)

askpatrickw commented:
@jepler I should have spelled that out. I was planning to do that in a separate PR. Fixing all those license files and weird pylint errors burned me out over lunch.

I think I'll flip this to draft this though and do it together.

askpatrickw commented:
OK, this is good to go.

- Updated build.yml to use pre-commit and removed no longer needed pip installs
- Moved pre-commit earlier in the order of steps so it will fail faster.
- fixed file name issues ubuntu is case-sensitive, macos\windows is not. 
- Going to leave the `from None` because I don't see this makefile and (make.py for windows) staying around unless we're also going to add it to the library workflow.

jepler reviewed (approved):
Thanks!
